qa/Rakefile

```diff
@@ -0,0 +1,66 @@
+require 'rspec'
+require 'rspec/core/rake_task'
+
+task :spec    => 'spec:all'
+task :default => :spec
+
+namespace :test do
+
+  def fetch_config
+    require "open3"
+    Open3.popen3("vagrant ssh-config") do |stdin, stdout, stderr, wait_thr|
+      { :stdout => stdout.read.chomp, :stderr => stderr.read.chomp,
+        :exit_status => wait_thr.value.exitstatus }
+    end
+  end
+
+  def parse(lines)
+    hosts, host = [], {}
+    lines.each do |line|
+      if line.match(/Host\s(.*)$/)
+        host = { :host => line.gsub("Host","").strip }
+      elsif line.match(/HostName\s(.*)$/)
+        host[:hostname] = line.gsub("HostName","").strip
+      elsif line.match(/Port\s(.*)$/)
+        host[:port]     = line.gsub("Port","").strip
+      elsif line.empty?
+        hosts << host
+        host = {}
+      end
+    end
+    hosts << host
+    hosts
+  end
+
+  desc "generate a valid ssh-config"
+  task :ssh_config do
+    require "json"
+    cd "acceptance" do
+      File.write("../.ssh_config", parse(fetch_config[:stdout].split("\n")).to_json)
+    end
+  end
+
+  namespace :acceptance do
+
+    desc "Run all acceptance"
+    task :all do
+      require "rspec/core/runner"
+      require "rspec"
+      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/**/*_spec.rb"]]))
+    end
+
+    desc "Run acceptance test in debian machines"
+    task :debian do
+      require "rspec/core/runner"
+      require "rspec"
+      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/debian/**/*_spec.rb"]]))
+    end
+
+    desc "Run acceptance test in centos machines"
+    task :centos do
+      require "rspec/core/runner"
+      require "rspec"
+      exit(RSpec::Core::Runner.run([Rake::FileList["acceptance/spec/centos/**/*_spec.rb"]]))
+    end
+  end
+end
```

Comment:
I am **suggesting** that we encapsulate the methods that deal with the `ssh_config` file into a class, so we can reuse it or unit test it more easily. 

IMHO  code in the rakefile should be minimal.